docs/src/advanced/conditional-requests.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Conditionally doing search requests
+mainTitle: Advanced
+layout: main.pug
+category: Advanced
+withHeadings: true
+navWeight: 7
+editable: true
+githubSource: docs/src/advanced/conditional-requests.md
+---
+
+> NOTE: this guide **has** been updated for v2
+
+InstantSearch will do a requests to Algolia servers on every key stroke, which means that it's fast and users will see updated results instantly. This also means that a request is being done to show the initial results on an empty query. While this usually isn't an issue and even a feature, since it warms up the network connection so subsequent requests will be done faster too.
+
+However, there are cases in which you do not want to do extra network requests than strictly necessary, for example when you are using a custom backend, or when the initial results are never displayed.
+
+## How it's made possible
+
+InstantSearch is the UI part on top a search client with a state managed by a [JavaScript Helper](https://github.com/algolia/algoliasearch-helper-js). These three layers are composable and can be interchanged to leverage the InstantSearch widgets system with different search clients.
+
+The [search client](https://github.com/algolia/algoliasearch-client-javascript) that Algolia offers queries Algolia's backends whenever the user refines the search. It is possible to implement your own search client that queries your backend, which then queries Algolia's backend with the [Algolia search client](https://github.com/algolia/algoliasearch-client-javascript) on your server.
+
+To create your own client, you will need to implement a given interface that receives and returns formatted data that InstantSearch can understand.
+
+## Implementing a proxy
+
+The way we will prevent searches to Algolia from happening in certain use cases is by making a proxy object around the `algoliasearch` client. This will then be passed to `ais-instant-search`:
+
+```vue
+<template>
+  <ais-instant-search
+    :search-client="searchClient"
+    index-name="instant_search"
+  >
+    <ais-search-box />
+    <ais-hits />
+  </ais-instant-search>
+</template>
+
+<script>
+import algoliasearch from "algoliasearch/lite";
+
+const algoliaClient = algoliasearch(
+  "latency",
+  "6be0576ff61c053d5f9a3225e2a90f76"
+);
+
+const searchClient = {
+  async search(requests) {
```

Comment:
Is `async` recommended with the common Vue tooling?

It is usually not easily compilable down to ES5.

Reply:
search is already returning a promise? 
In that case async is unneeded anyway.

Reply:
Regular vue tooling transpiles async indeed, reason I did is to avoid writing `Promise.resolve` in the interception code